import the data

In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

store_path = "../data/store.csv"
test_path = "../data/test.csv"
train_path = "../data/train.csv"

store = pd.read_csv(store_path)
test = pd.read_csv(test_path)
train = pd.read_csv(train_path)

take a first look at the data

In [5]:
train.describe()

Store       DayOfWeek           Sales       Customers  \
count  1017209.000000  1017209.000000  1017209.000000  1017209.000000   
mean       558.429727        3.998341     5773.818972      633.145946   
std        321.908651        1.997391     3849.926175      464.411734   
min          1.000000        1.000000        0.000000        0.000000   
25%        280.000000        2.000000     3727.000000      405.000000   
50%        558.000000        4.000000     5744.000000      609.000000   
75%        838.000000        6.000000     7856.000000      837.000000   
max       1115.000000        7.000000    41551.000000     7388.000000   

                 Open           Promo   SchoolHoliday  
count  1017209.000000  1017209.000000  1017209.000000  
mean         0.830107        0.381515        0.178647  
std          0.375539        0.485759        0.383056  
min          0.000000        0.000000        0.000000  
25%          1.000000        0.000000        0.000000  
50%          1.000000        0.000000        0.000000  
75%          1.000000        1.000000        0.000000  
max          1.000000        1.000000        1.000000

In [6]:
test.describe()

Id         Store     DayOfWeek          Open         Promo  \
count  41088.000000  41088.000000  41088.000000  41077.000000  41088.000000   
mean   20544.500000    555.899533      3.979167      0.854322      0.395833   
std    11861.228267    320.274496      2.015481      0.352787      0.489035   
min        1.000000      1.000000      1.000000      0.000000      0.000000   
25%    10272.750000    279.750000      2.000000      1.000000      0.000000   
50%    20544.500000    553.500000      4.000000      1.000000      0.000000   
75%    30816.250000    832.250000      6.000000      1.000000      1.000000   
max    41088.000000   1115.000000      7.000000      1.000000      1.000000   

       SchoolHoliday  
count   41088.000000  
mean        0.443487  
std         0.496802  
min         0.000000  
25%         0.000000  
50%         0.000000  
75%         1.000000  
max         1.000000

In [8]:
store.describe()

Store  CompetitionDistance  CompetitionOpenSinceMonth  \
count  1115.00000          1112.000000                 761.000000   
mean    558.00000          5404.901079                   7.224704   
std     322.01708          7663.174720                   3.212348   
min       1.00000            20.000000                   1.000000   
25%     279.50000           717.500000                   4.000000   
50%     558.00000          2325.000000                   8.000000   
75%     836.50000          6882.500000                  10.000000   
max    1115.00000         75860.000000                  12.000000   

       CompetitionOpenSinceYear       Promo2  Promo2SinceWeek  Promo2SinceYear  
count                761.000000  1115.000000       571.000000       571.000000  
mean                2008.668857     0.512108        23.595447      2011.763573  
std                    6.195983     0.500078        14.141984         1.674935  
min                 1900.000000     0.000000         1.000000      2009.000000  
25%                 2006.000000     0.000000        13.000000      2011.000000  
50%                 2010.000000     1.000000        22.000000      2012.000000  
75%                 2013.000000     1.000000        37.000000      2013.000000  
max                 2015.000000     1.000000        50.000000      2015.000000

In [9]:
train.head()

Store  DayOfWeek        Date  Sales  Customers  Open  Promo StateHoliday  \
0      1          5  2015-07-31   5263        555     1      1            0   
1      2          5  2015-07-31   6064        625     1      1            0   
2      3          5  2015-07-31   8314        821     1      1            0   
3      4          5  2015-07-31  13995       1498     1      1            0   
4      5          5  2015-07-31   4822        559     1      1            0   

   SchoolHoliday  
0              1  
1              1  
2              1  
3              1  
4              1

In [10]:
print len(train.Store.unique())
print len(train.Date.unique())

1115
942


In [11]:
test.head()

Id  Store  DayOfWeek        Date  Open  Promo StateHoliday  SchoolHoliday
0   1      1          4  2015-09-17     1      1            0              0
1   2      3          4  2015-09-17     1      1            0              0
2   3      7          4  2015-09-17     1      1            0              0
3   4      8          4  2015-09-17     1      1            0              0
4   5      9          4  2015-09-17     1      1            0              0

In [12]:
print len(test.Store.unique())
print len(test.Date.unique())

856
48


In [13]:
store.head()

Store StoreType Assortment  CompetitionDistance  CompetitionOpenSinceMonth  \
0      1         c          a                 1270                          9   
1      2         a          a                  570                         11   
2      3         a          a                14130                         12   
3      4         c          c                  620                          9   
4      5         a          a                29910                          4   

   CompetitionOpenSinceYear  Promo2  Promo2SinceWeek  Promo2SinceYear  \
0                      2008       0              NaN              NaN   
1                      2007       1               13             2010   
2                      2006       1               14             2011   
3                      2009       0              NaN              NaN   
4                      2015       0              NaN              NaN   

     PromoInterval  
0              NaN  
1  Jan,Apr,Jul,Oct  
2  Jan,Apr,Jul,Oct  
3              NaN  
4              NaN

In [14]:
print len(store.Store.unique())

1115


We have the historical data from 1115 store during 2013-1-1 to 2015-7-31 (942 days). And we try to predict the sales on 856 stores during 2015-8-1 to 2015-9-17 (48 days), totally 41008 entries.

train (1017209*9)
-- Store
-- DayOfWeek
-- Date
-- Sales
-- Customers
-- Open
-- Promo
-- StateHoliday
-- SchoolHoliday

test (41008*8)
-- Id
-- Store
-- DayofWeek
-- Date
-- Open
-- Promo
-- StateHoliday
-- SchoolHoliday

store
-- Store
-- StoreType (a,b,c,d)
-- Assortment (a = basic, b = extra, c = extended)
-- CompetitionDistance
-- CompetitionOpenSinceMonth
-- CompetitionOpenSinceYear
-- Promo2
-- Promo2SinceWeek
-- Promo2SinceYear
-- PromoInterval

In [16]:
train.isnull().sum()

Store            0
DayOfWeek        0
Date             0
Sales            0
Customers        0
Open             0
Promo            0
StateHoliday     0
SchoolHoliday    0
dtype: int64

In [17]:
test.isnull().sum()

Id                0
Store             0
DayOfWeek         0
Date              0
Open             11
Promo             0
StateHoliday      0
SchoolHoliday     0
dtype: int64

In [18]:
test.loc[ test.Open.isnull()]

Id  Store  DayOfWeek        Date  Open  Promo StateHoliday  \
479      480    622          4  2015-09-17   NaN      1            0   
1335    1336    622          3  2015-09-16   NaN      1            0   
2191    2192    622          2  2015-09-15   NaN      1            0   
3047    3048    622          1  2015-09-14   NaN      1            0   
4759    4760    622          6  2015-09-12   NaN      0            0   
5615    5616    622          5  2015-09-11   NaN      0            0   
6471    6472    622          4  2015-09-10   NaN      0            0   
7327    7328    622          3  2015-09-09   NaN      0            0   
8183    8184    622          2  2015-09-08   NaN      0            0   
9039    9040    622          1  2015-09-07   NaN      0            0   
10751  10752    622          6  2015-09-05   NaN      0            0   

       SchoolHoliday  
479                0  
1335               0  
2191               0  
3047               0  
4759               0  
5615               0  
6471               0  
7327               0  
8183               0  
9039               0  
10751              0

Found 11 NA in test data. They are all from store 622, and the dates are recent. A reasonable guess is that store 622 is not open anymore.